core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -1842,4 +1842,17 @@ private static RelDataType deriveTypeMapFromEntries(SqlOperatorBinding opBinding
           ReturnTypes.INTEGER_NULLABLE,
           OperandTypes.or(OperandTypes.CHARACTER, OperandTypes.BINARY),
           SqlFunctionCategory.NUMERIC);
+
+  /** The "BIT_GET(value, position)" function. */
+  @LibraryOperator(libraries = {SPARK})
+  public static final SqlBasicFunction BIT_GET =
+      SqlBasicFunction.create("BIT_GET",
+          ReturnTypes.INTEGER_NULLABLE,
```

Comment:
Are you sure that the return type is `INTEGER_NULLABLE`?

Reply:
According to Spark's [PR](https://github.com/apache/spark/pull/31198/files) to support `BIT_GET` function, the return type should be `TINYINT`. I will fix this soon.